src/audio/src/src.c

```diff
@@ -581,6 +581,10 @@ static int src_params(struct comp_dev *dev,
 	/* Set source/sink_rate/frames */
 	cd->source_rate = sourceb->stream.rate;
 	cd->sink_rate = sinkb->stream.rate;
+	if (cd->sink_rate == 0) {
+		comp_err(dev, "src_params(), zero sink rate");
+		return -EINVAL;
```

Comment:
ditto

Reply:
Like in a more fuzzy way described above I think this is the best way to handle streaming abort to prevent firmware crash.

When the rate converter filter is initialized in prepare() there's need to know the max frame size. Getting the output rate 0 Hz to fail similarly as request to initialize for 48000 Hz to 45678 Hz is not less code than this.